src/sinks/util/sink.rs

```diff
@@ -254,9 +260,15 @@ where
             partitions: HashMap::new(),
             timeout,
             lingers: HashMap::new(),
+            in_flight: None,
             closing: false,
         }
     }
+
+    /// Enforces per partition ordering of request.
+    pub fn ordered(&mut self) {
+        self.in_flight = Some(HashMap::new());
+    }
```

Comment:
I'm not particularly excited about having this behavior dynamically turned on at run time. How hard would it be to create a `struct OrderedPartitionBatchSink` as a wrapper around `PartitionBatchSink`, providing the ordering guarantee in  the outer layer?

Reply:
Nope. It can't be a layer above, nor a layer bellow. We can either separate this layer into separate pieces or raise ordering guarantee to type level like `ServiceLogic` is. `OrderingLogic`